File: RobustUnitTestingInAndroid/src/main/java/com/code/sliski/ui/api/StackoverflowApi.java
package com.code.sliski.ui.api;

import com.code.sliski.ui.model.PostWrapper;
import com.code.sliski.ui.model.Post;
import retrofit.Callback;
import retrofit.http.GET;
import retrofit.http.Path;

public interface StackoverflowApi {
    @GET("/users/{userId}/posts?site=stackoverflow")
    void getPosts(@Path("userId") long userId, Callback<PostWrapper<Post>> callback);
}

File: RobustUnitTestingInAndroid/src/main/kotlin/com/code/sliski/api/StackoverflowApi.kt
package com.code.sliski.api

import com.code.sliski.api.model.Post;
import com.code.sliski.api.model.PostWrapper
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface StackoverflowApi {

    @GET("users/{userId}/posts?site=stackoverflow")
    fun getPosts(@Path("userId") userId: Long): Single<PostWrapper<Post>>
}